File: c/move_zeroes.c
#include <stdbool.h>

void
moveZeroes(int *nums, int numsSize) {
	int z = 0;
	for (int i = 0; i < numsSize; i++) {
		while (z < numsSize && nums[z] != 0)
			z++;
		if (nums[i] != 0 && i > z) {
			nums[z] = nums[i];
			nums[i] = 0;
		}
	}
}

File: go/move_zeroes.go
package main

func moveZeroes(nums []int) {
	k := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[k], nums[i] = nums[i], nums[k]
			k++
		}
	}
}
